app/models/mixins/supports_feature_mixin.rb

```diff
@@ -116,6 +116,33 @@ def supports?(feature)
       public_send("supports_#{feature}?")
     end
 
+    # all subclasses that are considered for supporting features
+    def permitted_subclasses
+      descendants
+    end
+
+    def subclasses_supporting(feature)
+      permitted_subclasses.select { |subclass| subclass.supports?(feature) }
+    end
+
+    # Provider classes that support this feature
+    def provider_classes_supporting(feature)
+      subclasses_supporting(feature).map(&:module_parent)
+    end
```

Comment:
I'm not sure what the "provider" part of this method name means now (unless it's assuming that supports feature will only be mixed into "provider" related things (e.g. availability_zone, instances, etc), but _not_ non-provider things)

Reply:
this is "give me the ems classes that support this feature"

Reply:
This is the class version of the similar `providers_supporting` (which I think is more useful)

example:

Give me the Ems where I can create a network router:

```
  NetworkRouter.provider_supporting(:create) # => [Ems]
```